src/daemon/emitter.h

```diff
@@ -24,11 +24,12 @@ class Emitter : public CompilationDaemon {
     MESSAGE = 1,
     SOURCE = 2,
     EXTRA_FILES = 3,
+    HASH = 4,
```

Comment:
Same as in other places about hash -> shard:
It's a calculated hash value, and shard should be computed out of this and current number of shards(that may change).